File: static/code/sample/tethered/gdl-vithermal-a.cs
// In this sample:
// Read and display the temperature from pixel[0] to pixel[63]

using GHIElectronics.DUELink;

var availablePort = DUELinkController.GetConnectionPort();
var duelink = new DUELinkController(availablePort);

int GetTemperature(int pixel) {   
    return (int)duelink.Engine.ExecuteCommand($"PixTemp({pixel})");
}

while (true) {
    Console.WriteLine("==============================");
    for (var i = 0; i < 64; i++) {
        Console.WriteLine($"Temperature at pixel {i}: {GetTemperature(i)}");
    }

    Thread.Sleep(100);
}

File: static/code/sample/tethered/gdl-vithermal-a.py
# In this sample:
# Read and display the temperature from pixel[0] to pixel[63]

import time
from DUELink.DUELinkController import DUELinkController

availablePort = DUELinkController.GetConnectionPort()
duelink = DUELinkController(availablePort)

# method
def GetTemperature(pixel):
    return int(duelink.Engine.ExecuteCommand(f"PixTemp({pixel})"))

while True:
    print("==============================")
    for i in range(64):
        print(f"Temperature at pixel {i}: {GetTemperature(i)}")

    time.sleep(0.1)